interface.go

```diff
@@ -55,12 +55,52 @@ type Interface interface {
 	Stats() (*Stats, error)
 }
 
+type Datastore interface {
```

Comment:
I think this interface should be defined within the `store` directory and named `Interface`.  That way code imports the `"github.com/ipni/go-indexer-core/store"` package and refers to the store interface as `store.Interface`.

FYI: Throughout storetheindex, the terms "datastore" and "valuestore" refer to different things. The datastore stores indexer state data,  like ads processed, provider addrs, etc. The valuestore stores the index data (mapping of multihashes to provider values).

Reply:
Agreed - will move it over